src/Content/Widget/CalendarExport.php

```diff
@@ -0,0 +1,69 @@
+<?php
+
+/*
+ * @file src/Content/Widget/CalendarExport.php
+ */
+
+namespace Friendica\Content\Widget;
+
+use Friendica\Content\Feature;
+use Friendica\Core\L10n;
+
+require_once 'boot.php';
+require_once 'include/text.php';
+
+/**
+ * TagCloud widget
+ *
+ * @author Rabuzarus
+ */
+class CalendarExport
+{
+	/**
+	 * @brief Get the events widget.
+	 *
+	 * @return string Formated HTML of the calendar widget.
+	 */
+	public static function getHTML() {
+		$a = get_app();
+
+		$owner_uid = $a->data['user']['uid'];
+
+		// The permission testing is a little bit tricky because we have to respect many cases.
+
+		// It's not the private events page (we don't get the $owner_uid for /events).
+		if (!local_user() && !$owner_uid) {
+			return;
+		}
+
+		/*
+		 * If it's a kind of profile page (intval($owner_uid)) return if the user not logged in and
+		 * export feature isn't enabled.
+		 */
+		/*
+		 * Cal logged in user (test permission at foreign profile page).
+		 * If the $owner uid is available we know it is part of one of the profile pages (like /cal).
+		 * So we have to test if if it's the own profile page of the logged in user
+		 * or a foreign one. For foreign profile pages we need to check if the feature
+		 * for exporting the cal is enabled (otherwise the widget would appear for logged in users
+		 * on foreigen profile pages even if the widget is disabled).
+		 */
+		if (local_user() != $owner_uid && !Feature::isEnabled($owner_uid, "export_calendar")) {
```

Comment:
This check doesn't respect the case where `local_user()` didn't enabled the the `export_calendar` widget and visits his/her `/events` module. With your checks the widget would not appear now because `$owner_uid` would be `null`.

- `local_user()` should be able to see the widget for `/events` and his/her own `/cal` page (even if `export_calendar` is disabled)
- if `local_user()` is visiting a foreign profile page we have to respect if `export_calendar` is enabled (the situation on the local friendica instance)
- if a normal visitor is visiting a profile page we have to respect if `export_calendar` is enabled (the situation on foreign friendica instances)

Note: there seems to be a bug in the `/mod/events.php`. According to the code the widget should be loaded if `$a->argc > 1` but I think it should be loaded if `$a->argc >= 1`

Reply:
How do you access your own `/events` and `/cal` page? I had to manually manipulate the URL to reach the `/cal/[username]` page, and I still am not sure about the `/events` page. Additionally, with my new spreadsheet, I'm now sure this case wasn't covered before.

> Note: there seems to be a bug in the /mod/events.php. According to the code the widget should be loaded if `$a->argc > 1` but I think it should be loaded if `$a->argc >= 1`

`$a->argc` contains at least the module name (here, `events`), so the condition is correct.

Thanks for bearing with me!

Reply:
Can @annando answer while he's lurking around?

Reply:
Until yet I haven't watched this issue, since @rabuzarus seems to be active on this.

Reply:
I just need an answer to `How do you access your own /events and /cal page?`

Reply:
What is the problem with accessing these URLs? `/events` is in the navbar of the theme and `/cal` is for visitors of the profile. Or am I mistaken?

Reply:
- `/cal/[username]` can be accessed on someone else's profile but not on yourself. Widget is not shown. `$owner_uid` is set.
- `/events` is your personal calendar with the events you can see. Widget is not shown.
- `/events/[something]` is supposed to show the Widget, but I have no idea what `[something]` is supposed to be nor how to access it from the interface.

Reply:
AFAIK ```/events``` is only visible for yourself. ```/cal``` should be visible for everyone for public events and events that can be seen by the remote user.

Reply:
So it doesn't matter if `$owner_uid` is set or not, and the checks are correct.

Reply:
I don't know what happens when you visit ```/cal``` as a stranger and you click on a event. There could be called some ```/events``` folder.

Reply:
When a stranger tries to access my `/cal/hypolite` page, the query takes 517 seconds to complete, so I really can't tell you.

Reply:
Arghh, again. Lets say you are `local_user()` (let's say with `local_user() = 4) and you have disabled the widget (so value is `false`) than this check would lead to a return for `/events`.
`if (4 != null && false) {return;}`

BUT:
The `local_user()` should always see his/her own export widget on his/her own pages (`/events` and `/cal`) regardless of whether  the function has been enabled or not. The setting should be only for allowing visitors to see the widget or not.

With your current checks the `local_user()` can't see his/her own export widget if he/she has disabled it

Reply:
But the widget code isn't even called when `/events` is accessed without subsequent URI parameters, so the check isn't even evaluated in the first place.

On the other hand, while visiting `/cal/[own username]` the  `$owner_uid` is set and the widget correctly shows for self even with the feature disabled.

Reply:
> But the widget code has never been called when /events is accessed without subsequent URI parameters

yes this is a bug in `/mod/events.php`. Since `/cal` isn't easy accessible for the `local_user()`, the `local_user()` should have the possibility to export his/her own events under `/events`

Reply:
Finally, we got somewhere!

Reply:
Done.